### PR TITLE
Update composer.lock to prevent fixed bugs

### PR DESCRIPTION
**Pull request by mvorisek**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Various bugs were fixed, update locked versions to the latest. Pure lock file delete and fresh `composer install`. Source files are unmodified except CS needed to satisfy updated CS fixer.
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | all tests must pass

should be merged asap

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20716)

Updated versions:
```
beberlei/doctrineextensions: v1.2.6 => v1.2.7
composer/ca-bundle: 1.2.6 => 1.2.8
composer/installers: v1.7.0 => v1.9.0
doctrine/annotations: v1.8.0 => 1.10.4
doctrine/cache: 1.10.0 => 1.10.2
doctrine/collections: 1.6.4 => 1.6.7
doctrine/common: 2.12.0 => 2.13.3
doctrine/doctrine-bundle: 1.12.6 => 1.12.10
doctrine/event-manager: 1.1.0 => 1.1.1
doctrine/instantiator: 1.3.0 => 1.3.1
doctrine/orm: v2.7.0 => v2.7.3
doctrine/persistence: 1.3.5 => 1.3.8
doctrine/reflection: v1.1.0 => 1.2.1
egulias/email-validator: 2.1.15 => 2.1.19
ezyang/htmlpurifier: v4.12.0 => v4.13.0
fig/link-util: 1.1.0 => 1.1.1
friendsofsymfony/jsrouting-bundle: 2.5.3 => 2.6.0
incenteev/composer-parameter-handler: v2.1.3 => v2.1.4
markbaker/complex: 1.4.7 => 1.4.8
matthiasmullie/minify: 1.3.62 => 1.3.63
matthiasmullie/path-converter: 1.1.2 => 1.1.3
maxmind/web-service-common: v0.6.0 => v0.7.0
monolog/monolog: 1.25.3 => 1.25.5
nikic/php-parser: v4.3.0 => v4.9.0
phpoffice/phpspreadsheet: 1.10.1 => 1.12.0
prestashop/ps_featuredproducts: v2.0.0 => v2.1.0
psr/log: 1.1.2 => 1.1.3
react/promise: v2.7.1 => v2.8.0
symfony/http-client: v4.4.2 => v4.4.11
symfony/http-client-contracts: v1.1.8 => v1.1.9
symfony/mime: v4.4.2 => v4.4.11
symfony/polyfill-apcu: v1.13.1 => v1.18.1
symfony/polyfill-ctype: v1.13.1 => v1.18.1
symfony/polyfill-iconv: v1.13.1 => v1.18.1
symfony/polyfill-intl-icu: v1.13.1 => v1.18.1
symfony/polyfill-intl-idn: v1.13.1 => v1.18.1
symfony/polyfill-mbstring: v1.13.1 => v1.18.1
symfony/polyfill-php56: v1.13.1 => v1.18.1
symfony/polyfill-php70: v1.13.1 => v1.18.1
symfony/polyfill-php72: v1.13.1 => v1.18.1
symfony/polyfill-php73: v1.13.1 => v1.18.1
symfony/polyfill-util: v1.13.1 => v1.18.1
symfony/service-contracts: v1.1.8 => v1.1.9
symfony/symfony: v3.4.37 => v3.4.43
tecnickcom/tcpdf: 6.3.2 => 6.3.5
tijsverkoyen/css-to-inline-styles: 2.2.2 => 2.2.3
twig/twig: v1.42.4 => v1.43.1
ocramius/package-versions: n/a => 1.4.2
symfony/polyfill-intl-normalizer: n/a => v1.18.1
```

Code used to create this diff:
```
<?php

$lockOld = json_decode(<<<'EOD'

EOD
, true);

$lockNew = json_decode(<<<'EOD'

EOD
, true);

$toPackagesFunc = function ($lock) {
    $res = [];
    foreach ($lock['packages'] as $pkg) {
        $res[$pkg['name']] = $pkg;
    }
    return $res;
};

$pkgsOld = $toPackagesFunc($lockOld);
$pkgsNew = $toPackagesFunc($lockNew);

foreach (array_unique(array_merge(
    array_keys($pkgsOld),
    array_keys($pkgsNew),
)) as $n) {
    $vOld = ($pkgsOld[$n]['version'] ?? 'n/a');
    $vNew = ($pkgsNew[$n]['version'] ?? 'n/a');
    
    if ($vOld !== $vNew) {
        echo $n . ': '
            . $vOld . ' => ' . $vNew
            . "\n";
    }
}
```

### CODE REVIEW

**Comment by mvorisek** [comment]
All tests incl. Travis should be passing

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
You must split your pull request, many changes are not related to composer.lock
In addition, you also need to improve your pull request description to indicate which dependencies you update, and "from => to" which version :)

**Comment by mvorisek** [comment]
> 
> 
> You must split your pull request, many changes are not related to composer.lock
> In addition, you also need to improve your pull request description to indicate which dependencies you update, and "from => to" which version :)

> many changes are not related to composer.lock

every single change is related, what I did is:

- deleted lock file and recreated it using the latest deps matching current `composer.json` rules
- fixed CS with `cs-fixer fix` command (as manually reviewed/corrected the result)

@PierreRambaud so I would say there is nothing more done than locked deps update + fixes needed which deserves one PR, right?

**Comment by mvorisek** [comment]
> 
> 
> You must split your pull request, many changes are not related to composer.lock
> many changes are not related to composer.lock

every single change is related, what I did is:

- deleted lock file and recreated it using the latest deps matching current `composer.json` rules
- fixed CS with `cs-fixer fix` command (as manually reviewed/corrected the result)

do nothing more done than locked deps update + fixes needed which deserves one PR, right?

> In addition, you also need to improve your pull request description to indicate which dependencies you update, and "from => to" which version :)

@PierreRambaud done, can we merge? :)

**Comment by PierreRambaud** [comment]
image
I'm using the latest develop branch version and have nothing to change for cs-fixer. Are you sure you use `./vendor/bin/php-cs-fixer fix`, otherwise, you're maybe not using the right version.

**Comment by mvorisek** [comment]
please reread desc and see CI results for 1st commit - `php-cs-fixer` dep was also updated/fixed, thus some fixes were needed

**Comment by PierreRambaud** [comment]
> please reread desc and see CI results for 1st commit - `php-cs-fixer` dep was also updated/fixed, thus some fixes were needed

Didn't see the little text outside the table. I update your description :+1:

**Review by PierreRambaud — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -119,7 +119,7 @@ class FrontControllerCore extends Controller
     /** @var bool If true, forces display to maintenance page. */
     protected $maintenance = false;
 
-    /** @var string[] Adds excluded $_GET keys for redirection */
+    /** @var string[] Adds excluded _GET keys for redirection */
```
Comment (Progi1984) [inline classes/controller/FrontController.php]:
```suggestion
    /** @var string[] Adds excluded $_GET keys for redirection */
```

File: classes/helper/HelperList.php [hunk 2]
```diff
@@ -55,7 +55,7 @@ class HelperListCore extends Helper
     /** @var string ORDER BY clause determined by field/arrows in list header */
     public $orderBy;
 
-    /** @var string Default ORDER BY clause when $orderBy is not defined */
+    /** @var string Default ORDER BY clause when orderBy is not defined */
```
Comment (Progi1984) [inline classes/helper/HelperList.php]:
```suggestion
    /** @var string Default ORDER BY clause when $orderBy is not defined */
```

**Review by Progi1984 — COMMENTED** [review]
Some feedback

File: classes/controller/FrontController.php (hunk 1 above)
Comment (mvorisek) [reply classes/controller/FrontController.php]: yes, but CS fixer does not like it, what should I do? :)

File: classes/helper/HelperList.php (hunk 2 above)
Comment (mvorisek) [reply classes/helper/HelperList.php]: yes, but CS fixer does not like it, what should I do? :)

Comment (PierreRambaud) [reply classes/helper/HelperList.php]: It maybe needs to be escaped

Comment (mvorisek) [reply classes/helper/HelperList.php]: but how?

Comment (Progi1984) [reply classes/helper/HelperList.php]: with tilde "`"

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Qa is not required here.
Thanks @mvorisek

**Pull request merged**